crates/wash-lib/src/parser/mod.rs

```diff
@@ -510,3 +544,63 @@ impl RawProjectConfig {
         })
     }
 }
+
+impl ProjectConfig {
+    pub fn resolve_registry_url(&self) -> Result<String> {
+        let registry_url = match &self.common.registry.url {
+            Some(url) => url.clone(),
+            None => {
+                bail!("No registry URL specified in wasmcloud.toml");
+            }
+        };
+
+        Ok(registry_url)
+    }
+
+    pub async fn resolve_registry_credentials(
+        &self,
+        registry: impl AsRef<str>,
+    ) -> Result<RegistryAuth> {
+        let credentials_file = &self
+            .common
+            .registry
+            .credentials
+            .to_owned()
+            .or_else(|| default_registry_credentials_file().ok());
+
+        if credentials_file.is_none() {
+            return Ok(RegistryAuth::Anonymous);
+        }
+
+        let credentials_file = credentials_file.clone().unwrap();
+
+        if !credentials_file.exists() {
+            return Ok(RegistryAuth::Anonymous);
+        }
+
+        let credentials = tokio::fs::read_to_string(&credentials_file).await;
+        if credentials.is_err() {
+            return Ok(RegistryAuth::Anonymous);
+        }
+
+        let credentials =
+            serde_json::from_str::<RegistryCredentialMap>(&credentials.unwrap_or_default());
+        if credentials.is_err() {
+            return Ok(RegistryAuth::Anonymous);
```

Comment:
Consider printing/returning an error here, since this case — the file exists but is malformed — sounds like a user error